.github/workflows/main.yml

```diff
@@ -51,24 +51,41 @@ jobs:
                # Strip "v" prefix from tag name
                [[ "${{ github.ref }}" == "refs/tags/"* ]] && IMAGE_VERSION=$(echo $IMAGE_VERSION | sed -e 's/^v//')
                echo "IMAGE_VERSION=$IMAGE_VERSION" >> $GITHUB_ENV
-
+         
                echo "IMAGE_ID=$(echo ghcr.io/${{ github.repository_owner }}/$IMAGE_NAME | tr '[A-Z]' '[a-z]')" >> $GITHUB_ENV
 
-         # region See: https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#upgrading-a-workflow-that-accesses-ghcrio
-         -  name: Build Image
-            run: |
-               docker build -t $IMAGE_NAME .
+         # region See: https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#upgrading-a-workflow-that-accesses-ghcrio     
+         -  name: Use QEMU
+            uses: docker/setup-qemu-action@master
+            with:
+             platforms: all
+                      
+         - name: Free Disk Space (Ubuntu)
+           uses: insightsengineering/disk-space-reclaimer@v1
+           with:
+        # this might remove tools that are actually needed,
+        # if set to "true" but frees about 6 GB
+            tools-cache: true
+            android: true
+            dotnet: true
+            haskell: true
+            large-packages: true
+            swap-storage: true
+            docker-images: false
+      
+         - name: Set up Docker Buildx
+           id: buildx
+           uses: docker/setup-buildx-action@master
 
          -  name: Log In to Registry
             run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
 
-         -  name: Push Image
-            run: |
-               echo IMAGE_ID=$IMAGE_ID
-               echo IMAGE_VERSION=$IMAGE_VERSION
-               docker tag $IMAGE_NAME $IMAGE_ID:$IMAGE_VERSION
-               # Always tag latest when pushing a tag, as we don't expect to ever merge old tags
-               [[ "${{ github.ref }}" == "refs/tags/"* ]] && docker tag $IMAGE_NAME $IMAGE_ID:latest
-               docker push $IMAGE_ID:$IMAGE_VERSION
-               docker push $IMAGE_ID:latest
+         -  name: Build the docker image 
+            uses: docker/build-push-action@master
+            with:
+             context: .
+             builder: ${{ steps.buildx.outputs.name }}
+             platforms: linux/arm64,linux/amd64
+             push: true
+             tags: ghcr.io/kokomo123/5etools-img:latest               
```

Comment:
consider using e.g. `${{ github.repository_owner }}` to make this easier to test/fork/etc.

Reply:
Will do